float64 x
float64 y
float64 hash
string label